src/ScottPlot5/ScottPlot5/Primitives/SvgImage.cs

```diff
@@ -6,28 +6,42 @@ internal class SvgImage : IDisposable
     public readonly int Width;
     public readonly int Height;
     private readonly MemoryStream Stream;
-    public readonly SKCanvas Canvas;
+    public SKCanvas Canvas
+    {
+        get
+        {
+            // always create a new Canvas when required
+            if (_canvas is null && !IsDisposed)
+            {
+                SKRect rect = new(0, 0, Width, Height);
+                _canvas = SKSvgCanvas.Create(rect, Stream);
+            }
+            return _canvas!;
+        }
+    }
+    private SKCanvas? _canvas = null;
 
     public SvgImage(int width, int height)
     {
         Width = width;
         Height = height;
-        SKRect rect = new(0, 0, width, height);
         Stream = new MemoryStream();
-        Canvas = SKSvgCanvas.Create(rect, Stream);
     }
-
     public string GetXml()
     {
-        return Encoding.UTF8.GetString(Stream.ToArray()) + "</svg>";
+        // ensure Canvas is disposed before reading stream content to avoid errors
+        _canvas?.Dispose();
+        // Canvas no more relevant
+        _canvas = null; 
```

Comment:
This looks great, the only thing I might add is a link to the docs about why the canvas needs disposing. Maybe just something like this:

```cs
// The SVG document is not guaranteed to be closed until the canvas is disposed
// See: https://learn.microsoft.com/en-us/dotnet/api/skiasharp.sksvgcanvas?view=skiasharp-2.88#remarks
```